File: src/web/src/app/model/network-log.model.ts
export class NetworkLogModel {
  constructor() {
    this.logDate = null;
    this.rule = null;
    this.interfaceIn = null;
    this.interfaceOut = null;
    this.macAddress = null;
    this.sourceAddress = null;
    this.sourcePort = null;
    this.destinationAddress = null;
    this.destinationPort = null;
    this.packetLength = 0;
    this.protocol = null;
    this.timeToLive = 0;
    this.tcpAction = null;
    this.icmpSequence = null;
  }

  public logDate: Date | null;
  public rule: string | null;
  public interfaceIn: string | null;
  public interfaceOut: string | null;
  public macAddress: string | null;
  public sourceAddress: string | null;
  public destinationAddress: string | null;
  public packetLength: number;
  public timeToLive: number;
  public protocol: string | null;
  public sourcePort: number | null;
  public destinationPort: number | null;
  public tcpAction: string | null;
  public icmpSequence: number | null;
}

File: src/api/model/database/NetworkLogModel.cs
using System;

using MongoDB.Bson.Serialization.Attributes;

namespace Unionized.Model.Database
{
    public class NetworkLogModel : UnionizedModel
    {
        [BsonElement("logDate"), BsonRequired]
        public DateTime LogDate { get; set; }

        [BsonElement("rule"), BsonRequired]
        public string Rule { get; set; }

        [BsonElement("ifaceIn")]
        public string InterfaceIn { get; set; }

        [BsonElement("ifaceOut")]
        public string InterfaceOut { get; set; }

        [BsonElement("macAddr"), BsonRequired]
        public string MacAddress { get; set; }

        [BsonElement("sAddr"), BsonRequired]
        public string SourceAddress { get; set; }

        [BsonElement("dAddr"), BsonRequired]
        public string DestinationAddress { get; set; }

        [BsonElement("pLen"), BsonRequired]
        public int PacketLength { get; set; }

        [BsonElement("ttl"), BsonRequired]
        public int TimeToLive { get; set; }

        [BsonElement("proto"), BsonRequired]
        public string Protocol { get; set; }

        [BsonElement("sPrt")]
        public int? SourcePort { get; set; }

        [BsonElement("dPrt")]
        public int? DestinationPort { get; set; }

        [BsonElement("tcp")]
        public string TcpAction { get; set; }

        [BsonElement("icmpSeq")]
        public int? IcmpSequence { get; set; }
    }
}
